tools/model-registry/tests/test_smoke.py

```
@@ -469,3 +469,158 @@ def test_cli_classify_writes_class_to_gh_output(tmp_path, capsys, monkeypatch):
+    _, k_out, r_out = smoke.diagnose_chain(
+        load(tmp_path), "review", {"ZAI_KEY": "zk"}, _post_status(520, "outage"))
+    assert k_rl == k_out == "transient"
+    assert r_rl == "rate-limit" and r_out == "outage"
+
+
+def test_diagnose_network_error_reason_is_outage(tmp_path):
+    def post(url, headers, payload):
+        raise OSError("connection refused")
+    lines, klass, reason = smoke.diagnose_chain(load(tmp_path), "review", {"ZAI_KEY": "zk"}, post)
+    assert klass == "transient" and reason == "outage"
+    assert any("connection refused" in l for l in lines)
```

Ruff (0.16.2)

[error] Ambiguous variable name: `l`

(E741)